invenio_curations/requests/curation.py

```diff
@@ -29,27 +31,37 @@ def execute(self, identity, uow):
         receiver = self.request.receiver.resolve()
         record = self.request.topic.resolve()
 
-        # Adding a grant here as modifying the record permissions is quite tedious.
-        data = {
-            "grants": [
-                {
-                    "permission": "preview",
-                    "subject": {
-                        "type": "role",
-                        "id": str(receiver.id),
-                    },
-                    "origin": f"request:{self.request.id}",
-                }
-            ]
-        }
-
-        service = self.request.topic.get_resolver().get_service()
-        # NOTE: we're using the system identity here to avoid the grant creation
-        #       potentially being blocked by the requesting user's profile visibility
-        service.access.bulk_create_grants(system_identity, record.pid.pid_value, data)
-        uow.register(
-            ParentRecordCommitOp(record.parent, indexer_context=dict(service=service))
-        )
+        # if configured, share access to the record with moderators by creating grants
+        # rather than requiring an override of the record permission policy
+        if current_app.config.get("CURATIONS_PERMISSIONS_VIA_GRANTS", True):
```

Comment:
can we use the previously defined property of `current_curations` proxy here?

Reply:
hah, good catch!
this piece of code was older than the property which i just added yesterday :smile: